src/mlang/optimizing_ir/partial_evaluation.ml

```diff
@@ -411,16 +400,7 @@ let rec partially_evaluate_expr (ctx : partial_ev_ctx) (p : Mir.program)
                 (Bir_interpreter.evaluate_expr p
                    (Pos.same_pos_as (Mir.Unop (op, new_e1)) e1)
                    RegularFloat)
-          | _ -> (
-              ( Unop (op, new_e1),
-                match (op, d1) with
-                | Not, Float -> Float
-                | Not, Undefined -> Undefined
-                | Not, Top -> Top
-                | Minus, Float -> Float
-                | Minus, Undefined -> Float
-                | Minus, Top -> Float
```

Comment:
Are you sure about that? @rmonat

Reply:
To recap, it stated that `-x` was equivalent to `0-x`, i.e. always defined.
But I really doubted that and the legacy compiler agrees :
```
TSNV = positif (-TSNTV) * min (0 ,TSNTV+PRNNV)
       + positif_ou_nul (TSNTV) * TSNTV ;
```
This expects TSNTV to be defined in both side of the top addition. Which makes sense only if `positif(-TSNTV)` is defined iff the variable is.

Reply:
Ah indeed!